compressor/parser/beautifulsoup.py

```diff
@@ -1,42 +1,46 @@
 from __future__ import absolute_import
 from django.core.exceptions import ImproperlyConfigured
-from django.utils import six
 from django.utils.encoding import smart_text
 
-from compressor.exceptions import ParserError
 from compressor.parser import ParserBase
-from compressor.utils.decorators import cached_property
 
 
 class BeautifulSoupParser(ParserBase):
 
-    @cached_property
-    def soup(self):
+    def __init__(self, content):
+        super(BeautifulSoupParser, self).__init__(content)
         try:
-            if six.PY3:
-                from bs4 import BeautifulSoup
-            else:
+            from bs4 import BeautifulSoup
+            self.use_bs4 = True
+            self.soup = BeautifulSoup(self.content, "lxml")
```

Comment:
If "lxml" is ignored if not installed, then we can end up in the same situation the author depicts in that blog post, having a possibly different behaviour happen silently just because your environment changes.

For this reason I vote for using "html.parser" for now, just because it's always available.